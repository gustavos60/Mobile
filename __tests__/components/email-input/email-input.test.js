import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import EmailInput from '../../../src/components/email-input';

const mockedProps = {
  email: '',
  onChangeText: jest.fn(),
};

describe('email-input tests', () => {
  it('renders correctly', () => {
    const wrapper = renderer.create(<EmailInput {...mockedProps} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should have correct border color when invalid', () => {
    const wrapper = shallow(<EmailInput {...mockedProps} valid={false} />);
    const View = wrapper.find('View').first();
    const { style } = View.props();
    expect(style[1].borderColor).toEqual('#ff000A');
  });

  it('should have correct border color when valid', () => {
    const wrapper = shallow(<EmailInput {...mockedProps} valid />);
    const View = wrapper.find('View').first();
    const { style } = View.props();
    expect(style[1].borderColor).toEqual('gray');
  });
});
