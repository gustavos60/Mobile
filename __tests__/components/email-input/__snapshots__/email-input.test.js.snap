// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`email-input tests renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderColor": "gray",
        "borderRadius": 10,
        "borderWidth": 1,
        "flexDirection": "row",
        "margin": 10,
        "width": 600,
      },
      Object {
        "borderColor": "gray",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "gray",
          "fontSize": 30,
        },
        Object {
          "marginLeft": 10,
        },
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï‡®
  </Text>
  <TextInput
    allowFontScaling={true}
    autoCompleteType="email"
    keyboardType="email-address"
    onChangeText={[MockFunction]}
    placeholder="Email"
    rejectResponderTermination={true}
    style={
      Object {
        "flex": 1,
        "fontSize": 18,
        "padding": 10,
      }
    }
    testID=""
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
